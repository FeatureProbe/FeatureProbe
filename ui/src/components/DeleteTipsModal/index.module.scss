.modal-header-text {
  font-family: PingFangSC-Medium;
  font-size: 16px;
  color: rgba(0, 0, 0, 0.87);
}

.modal-content {
  font-size: 14px;
  color: rgba(0, 0, 0, 0.6);
  margin-top: 8px;
  margin-left: 40px;
  height: 20px;
  margin-bottom: 20px;
}

.warning-circle {
  font-size: 20px !important;
  color: #f5483b;
  padding: 0 10px;
}

.footer {
  text-align: right;
}

.btn {
  margin-right: 8px !important;
}
