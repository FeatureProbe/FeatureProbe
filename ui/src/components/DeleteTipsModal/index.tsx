import Icon from 'components/Icon';
import Modal from 'components/Modal';
import { ReactNode, SyntheticEvent } from 'react';
import { FormattedMessage } from 'react-intl';
import { Button, ButtonProps } from 'semantic-ui-react';

import styles from './index.module.scss';

interface DeleteTipsModalProps {
  open: boolean;
  onCancel: () => void;
  onConfirm: () => void;
  content: ReactNode;
  title: ReactNode;
  renderFooter?: (nodes: ButtonProps[]) => ButtonProps[];
}

const DeleteTipsModal: React.FC<DeleteTipsModalProps> = (props) => {
  const { open, onCancel, onConfirm, content, title, renderFooter } = props;

  return (
    <Modal
      open={open}
      width={400}
      footer={null}
    >
      <div>
        <div className={styles['modal-header']}>
          <Icon customclass={styles['warning-circle']} type="warning-circle" />
          <span className={styles['modal-header-text']}>
            {title}
          </span>
        </div>
        <div className={styles['modal-content']}>
          {content}
        </div>
        <div className={styles['footer']}>
          {(() => {
            const buttonProps: ButtonProps[] = [
              {
                size: 'mini',
                className: styles['btn'],
                type: 'reset',
                basic: true,
                onClick: (e: SyntheticEvent) => {
                  e.stopPropagation();
                  onCancel();
                },
                key: 'cancel',
                children: <FormattedMessage id="common.cancel.text" />
              },
              {
                size: 'mini',
                className: styles['btn'],
                type: 'submit',
                primary: true,
                onClick: (e: SyntheticEvent) => {
                  e.stopPropagation();
                  onConfirm();
                },
                key: 'confirm',
                children: <FormattedMessage id="common.confirm.text" />
              },
            ];
            const props = renderFooter ? renderFooter(buttonProps) : buttonProps;
            return props.map((prop) => {
              return <Button {...prop} />;
            });
          })()}
        </div>
      </div>
    </Modal>
  );
};

export default DeleteTipsModal;
