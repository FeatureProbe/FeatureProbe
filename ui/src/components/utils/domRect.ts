export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}