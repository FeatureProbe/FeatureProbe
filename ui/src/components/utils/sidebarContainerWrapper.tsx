import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
