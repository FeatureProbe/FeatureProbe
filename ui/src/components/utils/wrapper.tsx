import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
