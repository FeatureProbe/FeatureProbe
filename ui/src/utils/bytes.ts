export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;