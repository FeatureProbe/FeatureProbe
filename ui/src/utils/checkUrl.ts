function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}