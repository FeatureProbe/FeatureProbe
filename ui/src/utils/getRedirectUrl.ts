import { getProjectList } from 'services/project';
import { IProject, IEnvironment } from 'interfaces/project';
import { IDictionary } from 'interfaces/targeting';
import { getFromDictionary } from 'services/dictionary';
import { LAST_SEEN } from 'constants/dictionary_keys';

const getInfo = async () => {
  const res =  await getFromDictionary<IDictionary>(LAST_SEEN);
  const { success, data } = res;
  if (success && data) {
    return JSON.parse(data.value);
  } else {
    return {};
  }
};

export const getRedirectUrl = async (defaultRedirectUrl: string) => {
  let redirectUrl = defaultRedirectUrl;
  const info = await getInfo();
  const { projectKey, environmentKey } = info;

  if (!!projectKey && !!environmentKey) {
    redirectUrl = `/${projectKey}/${environmentKey}/toggles`;
  } else {
    const res = await getProjectList<IProject[]>();
    const { success, data } = res;
    if (success && data) {
      if (data[0]) {
        const { key, environments } = data[0];
        let envKey = '';
        environments.some((environment: IEnvironment) => {
          if (environment.key) {
            envKey = environment.key;
            return true;
          }
          return environment;
        });
  
        if (key && envKey) {
          redirectUrl = `/${key}/${envKey}/toggles`;
        }
      } else {
        redirectUrl = '/projects';
      }
    }
  }

  return redirectUrl;
};