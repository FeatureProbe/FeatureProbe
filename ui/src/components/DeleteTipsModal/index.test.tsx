import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import DeleteTipsModal from '.';
import { IntlWrapper } from 'components/utils/wrapper';

it('DeleteTipsModal snapshot', (done) => {
  (async () => {
    const { baseElement, rerender } = render(
      <DeleteTipsModal
        open={true}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
        content={'test content'}
        title={'test title'}
      />,
      {
        wrapper: IntlWrapper,
      }
    );
    expect(baseElement).toMatchSnapshot();

    rerender(
      <DeleteTipsModal
        open={true}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
        content={'test content'}
        title={'test title'}
        renderFooter={(nodes) => nodes}
      />
    );
    expect(baseElement).toMatchSnapshot();

    done();
  })();
});

it('DeleteTipsModal clcik', (done) => {
  (async () => {
    const mockOnCancel = jest.fn();
    const mockOnConfirm = jest.fn();
    render(
      <DeleteTipsModal
        open={true}
        onCancel={mockOnCancel}
        onConfirm={mockOnConfirm}
        content={'test content'}
        title={'test title'}
      />,
      {
        wrapper: IntlWrapper,
      }
    );
    
    await userEvent.click(screen.getByText('Confirm'));
    await userEvent.click(screen.getByText('Cancel'));

    expect(mockOnCancel).toBeCalled();
    expect(mockOnConfirm).toBeCalled();

    done();
  })();
});
