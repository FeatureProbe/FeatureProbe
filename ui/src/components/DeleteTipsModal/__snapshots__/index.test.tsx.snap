// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTipsModal snapshot 1`] = `
<body
  class="dimmable dimmed"
>
  <div />
  <div
    class="ui page modals dimmer transition visible active"
    style="display: flex;"
  >
    <div
      class="ui mini modal transition visible active"
      style="width: 400px;"
    >
      <div
        class="content"
      >
        <div>
          <div
            class="modal-header"
          >
            <span
              class="iconfont icon-warning-circle warning-circle"
              customclass="warning-circle"
              type="warning-circle"
            />
            <span
              class="modal-header-text"
            >
              test title
            </span>
          </div>
          <div
            class="modal-content"
          >
            test content
          </div>
          <div
            class="footer"
          >
            <button
              class="ui mini basic button btn"
              type="reset"
            >
              Cancel
            </button>
            <button
              class="ui mini primary button btn"
              type="submit"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`DeleteTipsModal snapshot 2`] = `
<body
  class="dimmable dimmed"
>
  <div />
  <div
    class="ui page modals dimmer transition visible active"
    style="display: flex;"
  >
    <div
      class="ui mini modal transition visible active"
      style="width: 400px;"
    >
      <div
        class="content"
      >
        <div>
          <div
            class="modal-header"
          >
            <span
              class="iconfont icon-warning-circle warning-circle"
              customclass="warning-circle"
              type="warning-circle"
            />
            <span
              class="modal-header-text"
            >
              test title
            </span>
          </div>
          <div
            class="modal-content"
          >
            test content
          </div>
          <div
            class="footer"
          >
            <button
              class="ui mini basic button btn"
              type="reset"
            >
              Cancel
            </button>
            <button
              class="ui mini primary button btn"
              type="submit"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
